packages/aws-cdk/lib/api/cxapp/exec.ts

```diff
@@ -40,8 +40,10 @@ export async function execProgram(aws: SdkProvider, config: Configuration): Prom
     context[cxapi.DISABLE_ASSET_STAGING_CONTEXT] = true;
   }
 
-  const bundlingStacks = config.settings.get(['bundlingStacks']) ?? ['*'];
-  context[cxapi.BUNDLING_STACKS] = bundlingStacks;
+  const bundlingStacks = config.settings.get(['bundlingStacks']);
+  if (bundlingStacks) {
+    context[cxapi.BUNDLING_STACKS] = bundlingStacks;
+  }
```

Comment:
An alternative to allowing `bundlingStacks` to be undefined when no pattern is supplied on the CLI (as in these changes) would be to set a valid "all" pattern for minimatch here (i.e. globstar `**` which matches stacks under a stage) and in settings.ts where the CLI input is parsed. Or handling `['*']` as a special case outside of minimatch (which the first commit does).

Reply:
I think I would be fine updating it to use `**` since I would imagine that was the intention from the start.

Reply:
I agree, I'd rather have fewer `if`s (fewer branches ⇒ fewer things that can behave unexpectedly).

So defaulting to `**` seems correct.

Reply:
Updated. It's worth noting that users explicitly specifying `*` on `cdk deploy '*'` will need to update it to `**` in order to match stacks under stages. Similarly, patterns like `*StackNameEnd` will need updating to `**/*StackNameEnd`.

Reply:
I actually think this is fine. Currently specifying `*` or `*StackNameEnd` does not work for stacks under a stage, you have to do `*/*` or `*/*StackNameEnd` which still work after this change.